File: 백준/Silver/2606. 바이러스/바이러스.js
const input = require("fs").readFileSync(process.platform === "linux" ? "/dev/stdin" : "알고리즘/example.txt").toString().trim().split("\n").map((el) => el.split(" ").map(Number));
const num_of_computer = input[0][0];
const connected_pair = input[1][0];
const pairs = input.slice(2);

// 그래프 그리기 시작 -> 0 ~ 7의 인덱스를 가질 수 있게 함
let graph = Array.from({ length: num_of_computer + 1 }, () => []);
for (let i = 0; i < connected_pair; i++) {
    let start = pairs[i][0];
    let end = pairs[i][1];
    graph[start].push(end);
    graph[end].push(start);
}

// dfs 시작
let visited = Array.from({ length: num_of_computer + 1 }, () => false);
let answer = 0;

function dfs(start_node) {
    // 시작 노드를 방문 표시
    visited[start_node] = true;

    // 시작 노드와 연결된 노드들에 대해 반복 (1 -> 2, 5 등)
    for (let end of graph[start_node]) {
        // 연결된 노드가 방문되지 않았다면 방문 표시하고 answer 증가
        if (!visited[end]) {
            visited[end] = true;
            answer++;
            // 연결된 노드에 대해 재귀적으로 dfs 수행
            dfs(end);
        }
    }
};

dfs(1);

// result는 감염된 컴퓨터 수 -> 1번 컴퓨터에 의해서 감염된 컴퓨터의 수이므로 -1
console.log(answer);

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
from collections import deque
input = sys.stdin.readline

num_of_computers = int(input())
connected_pair = int(input())
visited = [False] * (num_of_computers + 1)

# 인접 리스트로 그래프 표현
graph = {}
for _ in range(connected_pair):
    node1, node2 = map(int, input().split())
    
    # node1에 node2 추가
    if node1 in graph:
        graph[node1].append(node2)
    else:
        graph[node1] = [node2]
    
    # node2에 node1 추가
    if node2 in graph:
        graph[node2].append(node1)
    else:
        graph[node2] = [node1]

def check_virus_bfs(v, visited):
    queue = deque([v])
    visited[v] = True
    answer = 0
    
    while queue:
        node = queue.popleft()
        
        for neighbor in graph.get(node, []):
            if not visited[neighbor]:
                queue.append(neighbor)
                visited[neighbor] = True
                answer += 1
    return answer

print(check_virus_bfs(1, visited))